detox/runners/jest/reporters/DetoxReporter.js

```diff
@@ -6,19 +6,20 @@ const { config, reportTestResults } = require('../../../internals');
 
 class DetoxReporter extends JestVerboseReporter {
   /**
-   * @param {import('@jest/test-result').AggregatedResult} results
+   * @param {import('@jest/test-result').Test} test
+   * @param {import('@jest/test-result').TestResult} testResult
+   * @param {import('@jest/test-result').AggregatedResult} aggregatedResult
+   * @return {Promise<void>}
    */
-  // @ts-ignore
-  async onRunComplete(_contexts, results) {
-    // @ts-ignore
-    await super.onRunComplete(_contexts, results);
+  async onTestResult(test, testResult, aggregatedResult) {
+    await super.onTestResult(test, testResult, aggregatedResult);
 
-    await reportTestResults(results.testResults.map(r => ({
-      success: !r.failureMessage,
-      testFilePath: r.testFilePath,
-      testExecError: r.testExecError,
-      isPermanentFailure: this._isPermanentFailure(r),
-    })));
+    await reportTestResults([{
```

Comment:
This is also a bit unfinished thing – turned out, there was a great sense in reporting failed tests immediately. That helps to save log artifacts if the "--record-logs=failed".

Reply:
Described in https://github.com/wix/Detox/issues/3679 . Reverting.